plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableProperties.java

```diff
@@ -107,6 +123,35 @@ public IcebergTableProperties(
                         orcWriterConfig.getDefaultBloomFilterFpp(),
                         IcebergTableProperties::validateOrcBloomFilterFpp,
                         false))
+                .add(new PropertyMetadata<>(
+                        EXTRA_PROPERTIES,
+                        "Extra table properties",
+                        new MapType(VARCHAR, VARCHAR, typeManager.getTypeOperators()),
+                        Map.class,
+                        null,
+                        true, // currently not shown in SHOW CREATE TABLE
+                        value -> {
+                            Map<String, String> extraProperties = (Map<String, String>) value;
+                            if (extraProperties.containsValue(null)) {
+                                throw new TrinoException(INVALID_TABLE_PROPERTY, format("Extra table property value cannot be null '%s'", extraProperties));
+                            }
+                            if (extraProperties.containsKey(null)) {
+                                throw new TrinoException(INVALID_TABLE_PROPERTY, format("Extra table property key cannot be null '%s'", extraProperties));
+                            }
+
+                            Set<String> illegalExtraProperties = Sets.intersection(
```

Comment:
hmmm... on a second thought, we want to make sure that the user doesn't pass properties which are colliding with the iceberg native table properties and not with the Trino iceberg property names.

See again https://github.com/trinodb/trino/blob/292c101db17e77db7a8e475f85a04c797f1ad1ab/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L653-L673

This is likely where you want to make the overlapping check for properties.

Reply:
update accordingly the tests to check that the  user can't pass `write.format.default` through `extra_properties`.

Reply:
@findinpath What are your thoughts on validating only the Iceberg reserved properties? See https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/TableProperties.java#L74-L84.

I believe it would be difficult to maintain if we validate all the properties, and for every new Iceberg release, we would need to check for any new property and add it to our denylist/allowlist 🤔.

What are your thoughts?

Reply:
And I discovered that DEFAULT_FILE_FORMAT is not reserved. Maybe it should be? 🤔

Reply:
Check the Hive related PR for `extra_properties` . 
We don't have to disallow changing all the reserved Spark Iceberg properties.
We don't want to disallow changing Iceberg native table properties which relate to Trino Iceberg table properties.
e.g. `format` Trino table property translates to `write.format.default` native Iceberg property. We don't want to allow the user to side-track inserting `write.format.default` through `extra_properties`.

Reply:
@findinpath Done!